### PR TITLE
Font library: Fix React key prop warnings

### PR DESCRIPTION
**Pull request by mikachan**
## What?

Small PR to fix some unique key prop warnings that are currently happening when you open the Font Library modal.

## Testing Instructions

1. Open the Font Library modal, note there are no console warnings from this component.
2. Navigate between the tabs within the modal, note there are no console warnings here either.

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js [hunk 1]
```diff
@@ -76,7 +77,7 @@ function FontLibraryModal( {
 								contents = <InstalledFonts />;
 								break;
 							default:
-								contents = <FontCollection id={ id } />;
+								contents = <FontCollection />;
```
Comment (matiasbenedetto) [inline packages/edit-site/src/components/global-styles/font-library-modal/index.js]: The previous `id` now `slug` is needed in the  FontCollection component. It indicates the currently selected collection.

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js [hunk 2]
```diff
@@ -31,12 +31,13 @@ const DEFAULT_TABS = [
 ];
 
 const tabsFromCollections = ( collections ) =>
-	collections.map( ( { id, name } ) => ( {
-		id,
+	collections.map( ( { slug, name } ) => ( {
+		slug,
 		title:
-			collections.length === 1 && id === 'default-font-collection'
+			collections.length === 1 && slug === 'default-font-collection'
```
Comment (matiasbenedetto) [inline packages/edit-site/src/components/global-styles/font-library-modal/index.js]: This change was already added in the PR with refactor of the font collections: https://github.com/WordPress/gutenberg/pull/57884

File: packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js [hunk 3]
```diff
@@ -42,9 +42,13 @@ function FontsGrid( { title, children, pageSize = 32 } ) {
 				<div className="font-library-modal__fonts-grid__main">
 					{ items.map( ( child, i ) => {
 						if ( i === itemsLimit - 1 ) {
-							return <div ref={ setLastItem }>{ child }</div>;
+							return (
+								<div key={ i } ref={ setLastItem }>
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js]: It is [not recommended](https://react.dev/learn/rendering-lists#why-does-react-need-keys) to use the index as its key when rendering the list. Doesn't the child object have some kind of unique key instead?

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js (hunk 1 above)
Comment (mikachan) [reply packages/edit-site/src/components/global-styles/font-library-modal/index.js]: Ah I see, thanks! I've added it back here: https://github.com/WordPress/gutenberg/pull/57939/commits/22253e5a2fecccc5cf681e3e6d6769470527f74d.

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js (hunk 2 above)
Comment (mikachan) [reply packages/edit-site/src/components/global-styles/font-library-modal/index.js]: Gotcha, thanks. I've changed this so it's inline with that PR: https://github.com/WordPress/gutenberg/pull/57939/commits/c646e3d1b2039d763bcf3aae6141a3db6b712262.

File: packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js (hunk 3 above)
Comment (mikachan) [reply packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js]: Oh of course 🤦‍♀️ And there's a `key` on the child object! Updated here: https://github.com/WordPress/gutenberg/pull/57939/commits/2d3abce670f7ebdfb7650160427fa24775fbaf69. Thank you!

**Comment by mikachan** [comment]
I think the latest rebase has brought over some unwanted changes here, there should only be two files with changes:

- packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js
- packages/edit-site/src/components/global-styles/font-library-modal/style.scss

Happy for this to come in after we've finished the API refactor, no need to merge yet.

**Pull request merged**